test/marketing_event_test.rb

```diff
@@ -46,4 +46,23 @@ def test_count_marketing_events
     marketing_events_count = ShopifyAPI::MarketingEvent.get(:count)
     assert_equal 2, marketing_events_count
   end
+
+  def test_add_engagements
```

Comment:
I'm not sure this is an ideal refactor - mainly regarding the `assert`. This would be asserting the existence of the returned result of the post, correct? For testing purposes, if we got a 404, that assert would return true because it would have a returned response. 

I would understand loading the fixture instead, since that would be what the request body _would_ be. Any ideas on reaching a common ground @manasa91?